webui/react/src/pages/TrialDetails/TrialActions.tsx

```diff
@@ -43,11 +44,20 @@ const TrialActions: React.FC<Props> = ({ trial, onClick, onSettled }: Props) =>
     setButtonStates(state => ({ ...state, tensorboard: false }));
   }, [ trial.id, onSettled ]);
 
+  let trialCompletedCheckpointSum = 0;
+  experiment.trials.map((trial) => {
+    trialCompletedCheckpointSum += trial.numCompletedCheckpoints;
+  });
```

Comment:
**suggestion:** This is a good candidate for `useMemo` where you can memoize the result of an "expensive" operation.
```
const trialCompletedCheckpointSum = useMemo(() => {
  return trials.reduce((acc, trial) => acc + trial.numCompletedCheckpoints, 0);
}, [ trials ]);
```